server/src/main/java/org/elasticsearch/index/mapper/MappingLookup.java

```diff
@@ -220,6 +220,14 @@ FieldTypeLookup fieldTypesLookup() {
         return fieldTypeLookup;
     }
 
+    /**
+     * Returns the sum of all mappings, including field mappers (inclusive of alias fields and objects' flattened leaves), object mappers,
+     * and runtime fields.
+     */
+    public long getTotalMappings() {
+        return fieldMappers.size() + objectMappers.size() + runtimeFieldMappersCount;
```

Comment:
This looks right to me, I would rename it to `getTotalFieldsCount` and update the javadocs to say:

Returns the total number of fields defined in the mappings, including field mappers, object mappers as well as runtime fields.

Reply:
Maybe we could also make checkFieldLimit call this new method?

Reply:
Thank you @javanna , done.